lux/media/lux/lux.js

```diff
@@ -3430,58 +3447,427 @@ angular.module('lux.form.process', ['ngFileUpload'])
+                };
+
+                if (config.nameFromFormat) {
+                    parsedOption.repr = formatString(config.nameOpts.source, option);
+                }
+
+                if (attrs.multiple) {
+                    // For multiple field always get id of the object
+                    parsedOption.id = option.id;
+                }
+
+                return parsedOption;
+            },
+
+            /**
+             * Returns all parsed options.
+             *
+             * @param raw_options {object} - options fetched from the API
+             * @param options {object} - current `options` from select
+             * @param attrs {object}
+             * @param config {object} - parameters passed to query string in request
+             * @param extendCurrentOptions {boolean} - indicates whether to extend current `options` using `raw_options` from the API
+             */
+            getOptions: function(raw_options, options, attrs, config, extendCurrentOptions) {
+                angular.forEach(raw_options, function (option) {
+                    var parsedOption = remoteService.parseOption(option, attrs, config);
+
+                    if (extendCurrentOptions) {
+                        if (remoteService.excludeValue !== parsedOption.id) {
+                            options.push(parsedOption);
+                        }
+                    } else {
+                        options.push(parsedOption);
+                    }
+                });
+            }
+        };
+
+        return remoteService;
+    }])
+    /**
+     * Extension of ui-select to support infinite list of items.
+     */
+    .directive('selectInfinity', ['$parse', '$timeout', function($parse, $timeout) {
+        /**
+         * Returns the height of the given element
+         *
+         * @param elem
+         * @returns {integer} - height of the element
+         */
+        function height(elem) {
+            elem = elem[0] || elem;
+            if (isNaN(elem.offsetHeight)) {
+                return elem.document.documentElement.clientHeight;
+            } else {
+                return elem.offsetHeight;
+            }
+        }
+
+        /**
+         * Returns the distance from the top of the closest relatively positioned parent element.
+         *
+         * @param elem
+         * @returns {integer}
+         */
+        function offsetTop(elem) {
+            if (!elem[0].getBoundingClientRect || elem.css('none')) {
+                return;
+            }
+            return elem[0].getBoundingClientRect().top + pageYOffset(elem);
+        }
+
+        /**
+         * Returns the distance, in pixels, that a document has scrolled vertically.
+         *
+         * @param elem
+         * @returns {integer}
+         */
+        function pageYOffset(elem) {
+            elem = elem[0] || elem;
+            if (isNaN(window.pageYOffset)) {
+                return elem.document.documentElement.scrollTop;
```

Comment:
Similar question as the previous one here - what is this logic here intended to do? When is `elem.document` defined? Please add comments.